drivers/classic/src/calc_air_temperature.c

```diff
@@ -176,48 +174,56 @@ HourlyT(int     Dt,
 
 /******************************************************************************
  * @brief    This function estimates the times of minimum and maximum
- *           temperature for each day of the simulation, based on the hourly
+ *           temperature for each day of the simulation, based on the diurnal
  *           cycle of incoming solar radiation.
  *****************************************************************************/
 void
-set_max_min_hour(double *hourlyrad,
-                 int     ndays,
-                 int    *tmaxhour,
-                 int    *tminhour)
+set_max_min_sec(double *subdailyrad,
+                int     ndays,
+                int     stepsize,
+                int    *tmaxsec,
+                int    *tminsec)
 {
-    int risehour;
-    int sethour;
-    int hour;
-    int i;
+    size_t step;
+    int    risesec;
+    int    setsec;
+    int    sec;
+    int    i;
+    size_t stepsperday;
+
+    stepsperday = SEC_PER_DAY / stepsize;
```

Comment:
Does this require extra checking to make sure that `stepsize` evenly divides `SEC_PER_DAY`. If `stepsperday` is a quantity that is regularly used, it may make sense to add it to the time structure for VIC.